fenix/app/src/androidTest/java/org/mozilla/fenix/ui/SearchTest.kt

```diff
@@ -809,4 +809,52 @@ class SearchTest {
             }
         }
     }
+
+    // TestRail link: https://testrail.stage.mozaws.net/index.php?/cases/view/2348361
+    @SmokeTest
+    @Test
+    fun verifyFirefoxSuggestSponsoredSearchResultsTest() {
+        runWithCondition(
+            // This test will run only on Nightly and Debug builds as the feature is only available by default on these builds
+            activityTestRule.activity.components.core.engine.version.releaseChannel == EngineReleaseChannel.UNKNOWN ||
+                activityTestRule.activity.components.core.engine.version.releaseChannel == EngineReleaseChannel.NIGHTLY,
+        ) {
+            navigationToolbar {
+            }.clickUrlbar {
+                typeSearch(searchTerm = "Amazon")
+                verifySearchEngineSuggestionResults(
+                    rule = activityTestRule,
+                    searchSuggestions = arrayOf(
+                        "Firefox Suggest",
+                        "Amazon.com - Official Site",
+                        "Sponsored",
+                    ),
+                    searchTerm = "Amazon",
+                )
+            }
+        }
+    }
+
+    // TestRail link: https://testrail.stage.mozaws.net/index.php?/cases/view/2348374
+    @SmokeTest
+    @Test
+    fun verifyFirefoxSuggestNonSponsoredSearchResultsTest() {
+        runWithCondition(
+            // This test will run only on Nightly and Debug builds as the feature is only available by default on these builds
+            activityTestRule.activity.components.core.engine.version.releaseChannel == EngineReleaseChannel.UNKNOWN ||
+                activityTestRule.activity.components.core.engine.version.releaseChannel == EngineReleaseChannel.NIGHTLY,
+        ) {
+            navigationToolbar {
+            }.clickUrlbar {
+                typeSearch(searchTerm = "world cup")
+                verifySuggestionsAreNotDisplayed(
+                    rule = activityTestRule,
+                    searchSuggestions = arrayOf(
+                        "Firefox Suggest",
+                        "Sponsored",
+                    ),
+                )
+            }
+        }
+    }
```

Comment:
Just wondering if there's a feature flag that we can use here rather than check for channel?

Reply:
I think we could use either FeatureFlags.fxSuggest as a condition, or from the util class Settings, the enableFxSuggest var.